<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OutputParser</class>
 <widget class="QDialog" name="OutputParser">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>650</width>
    <height>582</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Output Parser </string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <layout class="QHBoxLayout" name="hlayoutPatternForReplacement">
       <property name="topMargin">
        <number>2</number>
       </property>
       <property name="bottomMargin">
        <number>2</number>
       </property>
       <item>
        <widget class="QLabel" name="labelPatternForReplacement">
         <property name="text">
          <string>Pattern to be replaced in output view log:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnAddPatternForReplacement">
         <property name="maximumSize">
          <size>
           <width>100</width>
           <height>20</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Add new search and replace pair</string>
         </property>
         <property name="text">
          <string>Add</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnModifyPatternForReplacement">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="maximumSize">
          <size>
           <width>100</width>
           <height>20</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Modify selected search and replace pair</string>
         </property>
         <property name="text">
          <string>Modify</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnRemovePatternForReplacement">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="maximumSize">
          <size>
           <width>100</width>
           <height>20</height>
          </size>
         </property>
         <property name="toolTip">
          <string>remove selected search and replace pair</string>
         </property>
         <property name="text">
          <string>Remove</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="hlayoutSearchReplace">
       <property name="topMargin">
        <number>2</number>
       </property>
       <property name="bottomMargin">
        <number>2</number>
       </property>
       <item>
        <widget class="QLabel" name="labelSearchPattern">
         <property name="text">
          <string>Search:  </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="edtSearchPattern">
         <property name="toolTip">
          <string>Enter search pattern that shall be replaced</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelReplacePattern">
         <property name="text">
          <string>Replace: </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="edtReplacePattern">
         <property name="toolTip">
          <string>Enter text to replace the search pattern</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QTableWidget" name="tablePatternForReplacement">
       <property name="toolTip">
        <string>Table of patterns for Search and replace within text output views</string>
       </property>
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="hlayoutPatternForMessage">
       <property name="topMargin">
        <number>2</number>
       </property>
       <property name="bottomMargin">
        <number>2</number>
       </property>
       <item>
        <widget class="QLabel" name="labelPatternForMessage">
         <property name="text">
          <string>Pattern to recognize message in output view log:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnAddPatternForMessage">
         <property name="maximumSize">
          <size>
           <width>100</width>
           <height>20</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Add new pattern for message recognition</string>
         </property>
         <property name="text">
          <string>Add</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnModifyPatternForMessage">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="maximumSize">
          <size>
           <width>100</width>
           <height>20</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Modify selected pattern for message recognition</string>
         </property>
         <property name="text">
          <string>Modify</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnRemovePatternForMessage">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="maximumSize">
          <size>
           <width>100</width>
           <height>20</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Remove selected pattern for message recognition</string>
         </property>
         <property name="text">
          <string>Remove</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QLineEdit" name="edtPattern">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Enter regular expression for recognition of messages: &lt;span style=&quot; font-style:italic;&quot;&gt;(.*)\(([\d]+)\): error: (.*)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="labelPositions">
       <property name="text">
        <string>Positions of placeholders:</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="hlayoutPlaceholderPositions">
       <property name="topMargin">
        <number>2</number>
       </property>
       <property name="bottomMargin">
        <number>2</number>
       </property>
       <item>
        <widget class="QLabel" name="labelFilenamePosition">
         <property name="text">
          <string>File name: </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinFilenamePosition">
         <property name="toolTip">
          <string>Select placeholder number within regular expression</string>
         </property>
         <property name="value">
          <number>1</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelLinePosition">
         <property name="text">
          <string>Line number</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinLinePosition">
         <property name="toolTip">
          <string>Select placeholder number within regular expression</string>
         </property>
         <property name="value">
          <number>2</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelMsgPosition">
         <property name="text">
          <string>Message text</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinMessageTextPositon">
         <property name="toolTip">
          <string>Select placeholder number within regular expression</string>
         </property>
         <property name="value">
          <number>3</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QLabel" name="labelTestMessageLine">
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>15</height>
        </size>
       </property>
       <property name="text">
        <string>Message line from output view as example to evaluate the pattern:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="edtTestMessageLine">
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Enter text from output view, that shall be recognized.&lt;/p&gt;&lt;p&gt;Example: &lt;span style=&quot; font-style:italic;&quot;&gt;/&lt;/span&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;home/user/project/main.cpp(41): error: undeclared identifier &amp;quot;i&amp;quot;&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="hlayoutFileName">
       <property name="topMargin">
        <number>2</number>
       </property>
       <property name="bottomMargin">
        <number>2</number>
       </property>
       <item>
        <widget class="QLabel" name="labelFileName">
         <property name="text">
          <string>File name: </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="edtFileName">
         <property name="toolTip">
          <string>Displays the file name of the text file from the output view log</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="hlayoutLineNumber">
       <property name="topMargin">
        <number>2</number>
       </property>
       <property name="bottomMargin">
        <number>2</number>
       </property>
       <item>
        <widget class="QLabel" name="labelLineNumber">
         <property name="text">
          <string>Line number: </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="edtLineNumber">
         <property name="toolTip">
          <string>Displays the line number of the output log message, if available</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="hlayoutMessageOutput">
       <property name="topMargin">
        <number>2</number>
       </property>
       <property name="bottomMargin">
        <number>2</number>
       </property>
       <item>
        <widget class="QLabel" name="labelMessageOutput">
         <property name="text">
          <string>Message text: </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="edtMessageOutput">
         <property name="toolTip">
          <string>Displays the text of the message from the output view</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QTableWidget" name="tablePatternForMessage">
       <property name="toolTip">
        <string>Table of patterns to recognize messages within output text views</string>
       </property>
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>OutputParser</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>OutputParser</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
